processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -714,14 +749,20 @@ public int getOffset()
     @Override
     public Offset clone()
     {
-      return new TimestampCheckingOffset(baseOffset.clone(), timestamps, threshold, allWithinThreshold);
+      return new TimestampCheckingOffset(baseOffset.clone(), timestamps, timeStart, timeEnd, allWithinThreshold);
     }
 
     @Override
     public boolean withinBounds()
     {
-      return baseOffset.withinBounds() && (allWithinThreshold
-                                           || timestamps.getLongSingleValueRow(baseOffset.getOffset()) < threshold);
+      if (!baseOffset.withinBounds()) {
+        return false;
+      }
+      if (allWithinThreshold) {
+        return true;
+      }
+      long current = timestamps.getLongSingleValueRow(baseOffset.getOffset());
+      return current >= timeStart && current < timeEnd;
```

Comment:
@navis we don't necessarily need to separate it out if it doesn't make a difference. My question was mainly whether we branch prediction would help us more than doing both checks, or maybe it doesn't make a difference at all, in which case we should leave the simplest code